netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -248,7 +252,16 @@ public void operationComplete(ChannelFuture future) throws Exception {
       keepAliveManager.onTransportStarted();
     }
 
-    return null;
+    return new Runnable() {
+      @Override
+      public void run() {
+        try {
+          listenerDone.sync();
```

Comment:
I suspect adding only the "flush" might be enough to make this test non flakey, given our 1second verification timeout. Let's try that.